### PR TITLE
HTSlib update to v1.6

### PR DESCRIPTION
**Pull request by mrphyja**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [ ] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

@bioconda/core <- this mention doesn't seem to be working?

First time contributor, but it's just a simple version update so not too much to break hopefully. I also also had to pin bzip2 to satisfy linting checks. Thanks!

### CODE REVIEW

File: recipes/pysam/meta.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ source:
 
 
 build:
-    number: 1
+    number: 2
```
Comment (bgruening) [inline recipes/pysam/meta.yaml]: I don't think you need this version bump. You can even reset it to 0 as you will get a new build string by increasing the htslib package in the env_matrix.

**Comment by bgruening** [comment]
ping @kyleabeauchamp for a review.

**Comment by mrphyja** [comment]
Maybe I should just revert that last commit for pysam and stick with just updating HTSlib? I could probably do another PR for pysam later if I end up needing it, it was kind of an afterthought anyway.

**Comment by kyleabeauchamp** [comment]
Yeah I would leave the pysam as is.

**Comment by kyleabeauchamp** [comment]
Otherwise lgtm

**Comment by mrphyja** [comment]
Ok, reverted the commit that changed pysam, just an HTSlib update now.

**Comment by bgruening** [comment]
@mrphyja better then reverting is to set it back to 0.

**Comment by mrphyja** [comment]
@bgruening I'm a bit confused, Kyle recommended to leave Pysam as is, so I basically just did a git revert on the commit that touched that file so now the only files being changed are `htslib/meta.yaml` and `env-matrix.yml`. Are you saying you think I should modify `pysam/meta.yaml`?

**Comment by bgruening** [comment]
If I remember correctly the pysam recipe has a build number of 1 right? This could be reset to 0 afaik as we get a new build string either way.

**Comment by mrphyja** [comment]
@bgruening Ok, thanks, I reset the build number to 0.

**Comment by bgruening** [comment]
Thanks @mrphyja!

**Pull request merged**